firestore/query.go

```diff
@@ -1101,24 +1234,33 @@ type DocumentIterator struct {
 // always concrete types, and the fact that this one has two different implementations
 // is an internal detail.
 type docIterator interface {
-	next() (*DocumentSnapshot, error)
+	next(opts ...RunOption) (*DocumentSnapshot, error)
+	getExplainMetrics() *ExplainMetrics
 	stop()
 }
 
 // Next returns the next result. Its second return value is iterator.Done if there
 // are no more results. Once Next returns Done, all subsequent calls will return
 // Done.
 func (it *DocumentIterator) Next() (*DocumentSnapshot, error) {
+	return it.nextWithOptions()
```

Comment:
Same question about if this iterator will be called with the same or different options every time. If it's the former, could we add a field in the `DocumentIterator` struct for a list of `RunOptions`?